From 3dd5cec57d2835e00e69a69f24fa60303216dbda Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <radostin@redhat.com>
Date: Sat, 4 Dec 2021 16:18:26 +0000
Subject: [PATCH 055/249] ci: Run cross compile on debian stable

The current debian stable release is Bullseye, not Buster. However, we
can use the 'stable' release instead. This would allow the CI to
automatically pick up updates in the future.

Signed-off-by: Radostin Stoyanov <radostin@redhat.com>
---
 .github/workflows/cross-compile-daily.yml                    | 2 +-
 .github/workflows/cross-compile.yml                          | 2 +-
 scripts/build/Dockerfile.aarch64-cross.tmpl                  | 1 -
 ...aarch64-cross.hdr => Dockerfile.aarch64-stable-cross.hdr} | 0
 scripts/build/Dockerfile.aarch64-stable-cross.tmpl           | 1 +
 scripts/build/Dockerfile.armv7-cross.tmpl                    | 1 -
 ...ile.armv7-cross.hdr => Dockerfile.armv7-stable-cross.hdr} | 0
 scripts/build/Dockerfile.armv7-stable-cross.tmpl             | 1 +
 scripts/build/Dockerfile.mips64el-cross.tmpl                 | 1 -
 ...ps64el-cross.hdr => Dockerfile.mips64el-stable-cross.hdr} | 0
 scripts/build/Dockerfile.mips64el-stable-cross.tmpl          | 1 +
 scripts/build/Dockerfile.ppc64-cross.tmpl                    | 1 -
 ...ile.ppc64-cross.hdr => Dockerfile.ppc64-stable-cross.hdr} | 0
 scripts/build/Dockerfile.ppc64-stable-cross.tmpl             | 1 +
 .../{Dockerfile.cross.tmpl => Dockerfile.stable-cross.tmpl}  | 5 ++---
 scripts/build/Makefile                                       | 2 +-
 16 files changed, 9 insertions(+), 10 deletions(-)
 delete mode 120000 scripts/build/Dockerfile.aarch64-cross.tmpl
 rename scripts/build/{Dockerfile.aarch64-cross.hdr => Dockerfile.aarch64-stable-cross.hdr} (100%)
 create mode 120000 scripts/build/Dockerfile.aarch64-stable-cross.tmpl
 delete mode 120000 scripts/build/Dockerfile.armv7-cross.tmpl
 rename scripts/build/{Dockerfile.armv7-cross.hdr => Dockerfile.armv7-stable-cross.hdr} (100%)
 create mode 120000 scripts/build/Dockerfile.armv7-stable-cross.tmpl
 delete mode 120000 scripts/build/Dockerfile.mips64el-cross.tmpl
 rename scripts/build/{Dockerfile.mips64el-cross.hdr => Dockerfile.mips64el-stable-cross.hdr} (100%)
 create mode 120000 scripts/build/Dockerfile.mips64el-stable-cross.tmpl
 delete mode 120000 scripts/build/Dockerfile.ppc64-cross.tmpl
 rename scripts/build/{Dockerfile.ppc64-cross.hdr => Dockerfile.ppc64-stable-cross.hdr} (100%)
 create mode 120000 scripts/build/Dockerfile.ppc64-stable-cross.tmpl
 rename scripts/build/{Dockerfile.cross.tmpl => Dockerfile.stable-cross.tmpl} (88%)

diff --git a/.github/workflows/cross-compile-daily.yml b/.github/workflows/cross-compile-daily.yml
index 701213276..927ddced2 100644
--- a/.github/workflows/cross-compile-daily.yml
+++ b/.github/workflows/cross-compile-daily.yml
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        target: [armv7-cross, aarch64-cross, ppc64-cross, mips64el-cross]
+        target: [armv7-stable-cross, aarch64-stable-cross, ppc64-stable-cross, mips64el-stable-cross]
         branches: [criu-dev, master]
 
     steps:
diff --git a/.github/workflows/cross-compile.yml b/.github/workflows/cross-compile.yml
index 90862e7ab..c6745d43e 100644
--- a/.github/workflows/cross-compile.yml
+++ b/.github/workflows/cross-compile.yml
@@ -8,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        target: [armv7-cross, aarch64-cross, ppc64-cross, mips64el-cross]
+        target: [armv7-stable-cross, aarch64-stable-cross, ppc64-stable-cross, mips64el-stable-cross]
 
     steps:
     - uses: actions/checkout@v2
diff --git a/scripts/build/Dockerfile.aarch64-cross.tmpl b/scripts/build/Dockerfile.aarch64-cross.tmpl
deleted file mode 120000
index 50eff9213..000000000
--- a/scripts/build/Dockerfile.aarch64-cross.tmpl
+++ /dev/null
@@ -1 +0,0 @@
-Dockerfile.cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.aarch64-cross.hdr b/scripts/build/Dockerfile.aarch64-stable-cross.hdr
similarity index 100%
rename from scripts/build/Dockerfile.aarch64-cross.hdr
rename to scripts/build/Dockerfile.aarch64-stable-cross.hdr
diff --git a/scripts/build/Dockerfile.aarch64-stable-cross.tmpl b/scripts/build/Dockerfile.aarch64-stable-cross.tmpl
new file mode 120000
index 000000000..81ef22980
--- /dev/null
+++ b/scripts/build/Dockerfile.aarch64-stable-cross.tmpl
@@ -0,0 +1 @@
+Dockerfile.stable-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.armv7-cross.tmpl b/scripts/build/Dockerfile.armv7-cross.tmpl
deleted file mode 120000
index 50eff9213..000000000
--- a/scripts/build/Dockerfile.armv7-cross.tmpl
+++ /dev/null
@@ -1 +0,0 @@
-Dockerfile.cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.armv7-cross.hdr b/scripts/build/Dockerfile.armv7-stable-cross.hdr
similarity index 100%
rename from scripts/build/Dockerfile.armv7-cross.hdr
rename to scripts/build/Dockerfile.armv7-stable-cross.hdr
diff --git a/scripts/build/Dockerfile.armv7-stable-cross.tmpl b/scripts/build/Dockerfile.armv7-stable-cross.tmpl
new file mode 120000
index 000000000..81ef22980
--- /dev/null
+++ b/scripts/build/Dockerfile.armv7-stable-cross.tmpl
@@ -0,0 +1 @@
+Dockerfile.stable-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.mips64el-cross.tmpl b/scripts/build/Dockerfile.mips64el-cross.tmpl
deleted file mode 120000
index 50eff9213..000000000
--- a/scripts/build/Dockerfile.mips64el-cross.tmpl
+++ /dev/null
@@ -1 +0,0 @@
-Dockerfile.cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.mips64el-cross.hdr b/scripts/build/Dockerfile.mips64el-stable-cross.hdr
similarity index 100%
rename from scripts/build/Dockerfile.mips64el-cross.hdr
rename to scripts/build/Dockerfile.mips64el-stable-cross.hdr
diff --git a/scripts/build/Dockerfile.mips64el-stable-cross.tmpl b/scripts/build/Dockerfile.mips64el-stable-cross.tmpl
new file mode 120000
index 000000000..81ef22980
--- /dev/null
+++ b/scripts/build/Dockerfile.mips64el-stable-cross.tmpl
@@ -0,0 +1 @@
+Dockerfile.stable-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.ppc64-cross.tmpl b/scripts/build/Dockerfile.ppc64-cross.tmpl
deleted file mode 120000
index 50eff9213..000000000
--- a/scripts/build/Dockerfile.ppc64-cross.tmpl
+++ /dev/null
@@ -1 +0,0 @@
-Dockerfile.cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.ppc64-cross.hdr b/scripts/build/Dockerfile.ppc64-stable-cross.hdr
similarity index 100%
rename from scripts/build/Dockerfile.ppc64-cross.hdr
rename to scripts/build/Dockerfile.ppc64-stable-cross.hdr
diff --git a/scripts/build/Dockerfile.ppc64-stable-cross.tmpl b/scripts/build/Dockerfile.ppc64-stable-cross.tmpl
new file mode 120000
index 000000000..81ef22980
--- /dev/null
+++ b/scripts/build/Dockerfile.ppc64-stable-cross.tmpl
@@ -0,0 +1 @@
+Dockerfile.stable-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.cross.tmpl b/scripts/build/Dockerfile.stable-cross.tmpl
similarity index 88%
rename from scripts/build/Dockerfile.cross.tmpl
rename to scripts/build/Dockerfile.stable-cross.tmpl
index 8b95fbb1c..6a68cd1ca 100644
--- a/scripts/build/Dockerfile.cross.tmpl
+++ b/scripts/build/Dockerfile.stable-cross.tmpl
@@ -1,9 +1,8 @@
 COPY scripts/ci/apt-install /bin/apt-install
 
 # Add the cross compiler sources
-RUN echo "deb http://deb.debian.org/debian/ buster main" >> /etc/apt/sources.list && \
-  dpkg --add-architecture ${DEBIAN_ARCH} && \
-  apt-install emdebian-archive-keyring
+RUN echo "deb http://deb.debian.org/debian/ stable main" >> /etc/apt/sources.list && \
+  dpkg --add-architecture ${DEBIAN_ARCH}
 
 RUN apt-install \
 	crossbuild-essential-${DEBIAN_ARCH}	\
diff --git a/scripts/build/Makefile b/scripts/build/Makefile
index 62e3a9920..a436c2839 100644
--- a/scripts/build/Makefile
+++ b/scripts/build/Makefile
@@ -1,5 +1,5 @@
 ARCHES := x86_64 fedora-asan fedora-rawhide centos7 armv7hf centos8
-NON_CLANG := armv7-cross aarch64-cross ppc64-cross mips64el-cross
+NON_CLANG := armv7-stable-cross aarch64-stable-cross ppc64-stable-cross mips64el-stable-cross
 CREATE_DOCKERFILES := $(ARCHES) $(NON_CLANG)
 TARGETS := $(ARCHES) alpine archlinux
 TARGETS_CLANG := $(addsuffix $(TARGETS),-clang)
-- 
2.35.1

